lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,256 @@
+#
+# [source,ruby]
+
+#
+module LogStash module Filters class JdbcStatic < LogStash::Filters::Base
+  config_name "jdbc_static"
+
+  # Define the set (Hash) of loaders to fetch remote data and create local tables.
+  # the fetched data will be inserted into the local tables. Make sure that the
+  # local table name, columns and datatypes correspond to the shape of the remote data
+  # being fetched.
+  # You can also specify alternate JDBC strings to use if there is more than one remote database
+  # For example:
+  # loaders => [
+  #   {
+  #     "id" => "country_details"
+  #     "query" => "select code, name from WORLD.COUNTRY"
+  #     "max_rows" => 2000
+  #     local_table => "country"
+  #   },
+  #   {
+  #     id => "servers_load"
+  #     query => "select id, ip, name, location from INTERNAL.SERVERS"
+  #     local_table => "servers"
+  #   }
+  # ]
+  # This is optional. You can provide a pre-populated local database server then no initial loaders are needed.
+  config :loaders, :required => true, :default => [], :validate => [LogStash::Filters::Util::Loader]
+
+  # Define an array of Database Objects to create when the plugin first starts.
+  # These will usually be the definitions to setup the local in-memory tables.
+  # For example:
+  # local_db_objects => [
+  #   {type => "table", name => "servers", columns => [["id", "INTEGER"], ["ip", "varchar(64)"], ["name", "varchar(64)"], ["location", "varchar(64)"]]},
+  #   {type => "index", name => "servers_idx", table => "servers", columns => ["ip"]}
+  # ]
+  # NOTE: Important! Tables created here must have the same names as those used in the `loaders` and
+  # `lookups` configuration options
+  config :local_db_objects, :required => false, :default => [], :validate => [LogStash::Filters::Util::DbObject]
+
+  # Define the list (Array) of enhancement lookups to be applied to an event
+  # Each entry is a hash of the query string, the target field and value and a
+  # parameter hash. Target is overwritten if existing. Target is optional,
+  # if omitted the lookup results will be written to the root of the event like this:
+  # event.set(<column name (or alias)>, <column value>)
+  # Use parameters to have this plugin put values from the event into the query.
+  # The parameter maps the symbol used in the query string to the field name in the event.
+  # NOTE: when using a query string that includes the LIKE keyword make sure that
+  # you provide a Logstash Event sprintf pattern with added wildcards.
+  # For example:
+  # lookups => [
+  #   {
+  #     "query" => "select * from country WHERE code = :code",
+  #     "parameters" => {"code" => "country_code"}
+  #     "target" => "country_details"
+  #   },
+  #   {
+  #     "query" => "select ip, name from servers WHERE ip LIKE :ip",
+  #     "parameters" => {"ip" => "%{[response][ip]}%"}
+  #     "target" => "servers"
+  #   }
+  # ]
+  config :lookups, :required => true, :validate => [LogStash::Filters::Util::Lookup]
```

Comment:
Can we change this to be `local_lookups`. That way it is self-descriptive we are doing a local lookup, not remote..

Reply:
done